src/matching/quoteMatcher.ts

```diff
@@ -0,0 +1,36 @@
+/**
+ * QuoteMatcher matches quoted strings, respecting escaped quotes (\") and frieds
+ */
+export class QuoteMatcher {
+  static escapeChar = "\\";
+
+  static findOpening(char: string, corpus: string, start: number): number {
+    // first, search backwards to see if we could be inside a quote
+    for (let i = start - 1; i >= 0; i--) {
+      const currentChar = corpus[i];
+      if (currentChar === char && corpus[i - 1] !== this.escapeChar) {
+        return i;
+      }
+    }
+
+    // Didn't find one behind us, the string may start ahead of us
+    for (let i = start; i < corpus.length; i++) {
+      const currentChar = corpus[i];
+      if (currentChar === char && corpus[i - 1] !== this.escapeChar) {
```

Comment:
We have handled escape char while parsing subsitute commands https://github.com/VSCodeVim/Vim/blob/master/src/cmd_line/subparsers/substitute.ts . The logic is reading char one by one and handling every `\`. I suppose we can do the same thing here?

For now I don't see any other case but I'll investigate.